movies/3.api-doc.md

```diff
@@ -0,0 +1,89 @@
+
+## API behaviour
+
+```http
+GET https://hydra-movies.herokuapp.com/
+```
+
+Response headers (excerpt):
+
+```http
+Link: </doc>; rel="http://www.w3.org/ns/hydra/core#apiDocumentation"
+```
+
+```http
+GET https://hydra-movies.herokuapp.com/doc
+```
+
+Response body (excerpt):
+
+```json
+{
+  "@context": "/context.jsonld",
+  "@id": "/doc",
+  "@type": "ApiDocumentation",
+  "title": "Hydra Movies API",
+  "description": "This API shows how a basic CRUD (Create, Read, Update, Delete) API can be designed using Hydra Core Vocabulary.",
+  "entrypoint": "/",
+  "supportedClass": [
+    {
+      "@id": "schema:EntryPoint",
+      "@type": "Class",
+      "title": "API entrypoint",
+      "description": "Every capability of the API can be reached from here by following links."
+    }
+  ]
+}
+```
+
+### Explanation
+
+```http
+Link: </doc>; rel="http://www.w3.org/ns/hydra/core#apiDocumentation"
+```
+
+```http
+GET https://hydra-movies.herokuapp.com/doc
+```
+
+As described in ["API entrypoint"](./1.entry-point.md), every API response has to contain a Link to the documentation. This allows a client to dereference and process the documentation, even if it does not start from the entrypoint (e.g. because of a "deep link" or a bookmark from a former visit).
+
+```
+"@context": "/context.jsonld",
+"@id": "/doc",
+"@type": "ApiDocumentation"
+```
+
+The API documenation is just another web rescoure, with of course an `@id` (i.e. URI) and the type `http://www.w3.org/ns/hydra/core#ApiDocumentation`, which is abbreviated to `ApiDocumentation` here thanks to the JSON-LD context.
+
+```
+"title": "Hydra Movies API",
+"description": "This API shows how a basic CRUD (Create, Read, Update, Delete) API can be designed using Hydra Core Vocabulary.",
+```
+
+`title` and `description` are hydra terms that allow to describe the API in a human-readable matter on a high level. The values provided here, could for example appear in generic clients, API catalogs or API search engine results. They allow humans to get a first impression of what the API is good for.
+
+```
+"entrypoint": "/",
+```
+
+Since a client may get to the API by any URL, not only the entry point, it is important to make clear what the entry point of the API actually is. The entry point may also differ by the state of a client: An authenticated client could for example be sent to a user profile, while anonymous clients get a generic entry point. 
+
+```
+"supportedClass": [
+    {
+      "@id": "schema:EntryPoint",
+      "@type": "Class",
+      "title": "API entrypoint",
+      "description": "Every capability of the API can be reached from here by following links."
+    }
+  ]
+```
+
+The hydra term `supportedClass` describes all classes the API provides or is able to handle in some way. The most basic thing we can describe here, is again a human-readble `title` and `description`. More possibilities will be shown in later recipes.
```

Comment:
I'm not sure `all classes` is correct. I believe it may vary when the state changes and the initial list may not contain all classes.

Reply:
This statement isn't factually incorrect but @alien-mcl has a point.

How about adding the [hints plugin](https://github.com/GitbookIO/plugin-hints) and add an emphasised remark clarifying that both the entrypoint and the Api Documentation can vary between the states of client and server.

Reply:
sounds good to me